packages/outstatic/src/pages/add-custom-field/index.tsx

```diff
@@ -0,0 +1,557 @@
+
+type AddCustomFieldProps = {
+  collection: string
+}
+
+type CustomFieldForm = CustomField & { name: string }
+
+export default function AddCustomField({ collection }: AddCustomFieldProps) {
+  const {
+    contentPath,
+    monorepoPath,
+    session,
+    repoSlug,
+    repoBranch,
+    repoOwner
+  } = useContext(OutstaticContext)
+  const [createCommit] = useCreateCommitMutation()
+  const fetchOid = useOid()
+  const [hasChanges, setHasChanges] = useState(false)
+  const [customFields, setCustomFields] = useState<CustomFields>({})
+  const yupSchema = yup.object().shape({
+    title: yup
+      .string()
+      .matches(/^[aA-zZ\s]+$/, 'Only alphabets are allowed for this field.')
+      .required('Custom field name is required.'),
+    type: yup.string().oneOf(['text', 'string']).required(),
+    description: yup.string(),
+    required: yup.boolean().required()
+  })
+  const [adding, setAdding] = useState(false)
+  const [error, setError] = useState('')
+  const methods = useForm<CustomFieldForm>({
+    mode: 'onChange',
+    resolver: yupResolver(yupSchema)
+  })
+  const [showAddModal, setShowAddModal] = useState(false)
+  const [showDeleteModal, setShowDeleteModal] = useState(false)
+  const [deleting, setDeleting] = useState(false)
+  const { data: schemaQueryData, loading } = useFileQuery({
+    file: `${collection}/schema.json`
+  })
+  const [selectedField, setSelectedField] = useState('')
+  const [fieldName, setFieldName] = useState('')
+
+  const capiHelper = async ({
+    customFields,
+    deleteField = false
+  }: {
+    customFields: any
+    deleteField?: boolean
+  }) => {
+    const oid = await fetchOid()
+    const customFieldsJSON = JSON.stringify(
+      {
+        title: collection,
+        type: 'object',
+        properties: { ...customFields }
+      },
+      null,
+      2
+    )
+
+    const capi = createCommitApi({
+      message: `feat(${collection}): ${
+        deleteField ? 'delete' : 'add'
+      } ${fieldName} field`,
+      owner: repoOwner || session?.user?.login || '',
+      oid: oid ?? '',
+      name: repoSlug,
+      branch: repoBranch
+    })
+
+    capi.replaceFile(
+      `${
+        monorepoPath ? monorepoPath + '/' : ''
+      }${contentPath}/${collection}/schema.json`,
+      customFieldsJSON + '\n'
+    )
+
+    const input = capi.createInput()
+
+    return await createCommit({ variables: { input } })
+  }
+
+  const onSubmit: SubmitHandler<CustomFieldForm> = async (
+    data: CustomFieldForm
+  ) => {
+    setAdding(true)
+    const { title, ...rest } = data
+    const fieldName = camelCase(title)
+
+    if (!selectedField && customFields[fieldName]) {
+      methods.setError('title', {
+        type: 'manual',
+        message: 'Field name is already taken.'
+      })
+      setAdding(false)
+      return
+    }
+
+    try {
+      customFields[fieldName] = {
+        ...rest,
+        title: capitalize(data.title)
```

Comment:
Should this be done in CSS instead? https://tailwindcss.com/docs/text-transform#transforming-text